assembly/index.ts

```diff
@@ -165,40 +158,34 @@ export function antialiased(img: Uint8Array, x1: i32, y1: i32, width: i32, heigh
 	}
 
 	// if there are no both darker and brighter pixels among siblings, it's not anti-aliasing
-	if (min === 0 || max === 0) {
+	if (i32(min === 0) | i32(max === 0)) {
 		return false;
 	}
 
 	// if either the darkest or the brightest pixel has 3+ equal siblings in both images
 	// (definitely not anti-aliased), this pixel is anti-aliased
 	return (
-		hasManySiblings(img, minX, minY, width, height) &&
-		hasManySiblings(img2, minX, minY, width, height)
-	) || (
-		hasManySiblings(img, maxX, maxY, width, height) && hasManySiblings(img2, maxX, maxY, width, height)
-	);
+    (hasManySiblings(imgPtr, minX, minY, width, height) && hasManySiblings(img2Ptr, minX, minY, width, height)) ||
+    (hasManySiblings(imgPtr, maxX, maxY, width, height) && hasManySiblings(img2Ptr, maxX, maxY, width, height))
+  );
 }
 
 // // check if a pixel has 3+ adjacent pixels of the same color.
-export function hasManySiblings(img: Uint8Array, x1: i32, y1: i32, width: i32, height: i32): bool {
-	const x0 = Math.max(x1 - 1, 0);
-	const y0 = Math.max(y1 - 1, 0);
-	const x2 = Math.min(x1 + 1, width - 1);
-	const y2 = Math.min(y1 + 1, height - 1);
-	const pos = (y1 * width + x1) * 4;
-	let zeroes = x1 === x0 || x1 === x2 || y1 === y0 || y1 === y2 ? 1 : 0;
+export function hasManySiblings(imgPtr: usize, x1: i32, y1: i32, width: i32, height: i32): bool {
+	let x0 = max(x1 - 1, 0);
+	let y0 = max(y1 - 1, 0);
+	let x2 = min(x1 + 1, width - 1);
+	let y2 = min(y1 + 1, height - 1);
+	let pos = (y1 * width + x1) * 4;
+	let zeroes = i32(x1 === x0) | i32(x1 === x2) | i32(y1 === y0) | i32(y1 === y2);
 
 	// go through 8 adjacent pixels
-	for (let x = x0; x <= x2; x++) {
-		for (let y = y0; y <= y2; y++) {
-			if (x === x1 && y === y1) continue;
-
-			const pos2: i32 = ((y * width + x) * 4) as i32
-			if (img[pos] === img[pos2] &&
-					img[pos + 1] === img[pos2 + 1] &&
-					img[pos + 2] === img[pos2 + 2] &&
-					img[pos + 3] === img[pos2 + 3]) zeroes++;
-
+  for (let y = y0; y <= y2; y++) {
+    let stride = y * width << 2;
+	  for (let x = x0; x <= x2; x++) {
+			if (i32(x === x1) & i32(y === y1)) continue;
+			let pos2 = stride + (x << 2);
+      zeroes += i32(load<u32>(imgPtr + pos) == load<u32>(imgPtr + pos2));
```

Comment:
Does `imgPtr + pos` cast `pos` to `usize`? Might like to wrap `pos` and `pos2`  in `usize()` because 64 bit wasm reasons. I don't know what standard operating procedure is here. :)